modules.d/99suse-initrd/module-setup.sh

```diff
@@ -22,14 +22,23 @@ get_modprobe_conf_files() {
 # called by dracut
 installkernel() {
     local line mod reqs all_mods=
+    local BUILT_IN_PATH="/lib/modules/$(uname -r)/modules.builtin"
 
     while read -r line; do
         mod="${line##*SUSE INITRD: }"
         mod="${mod%% REQUIRES*}"
         reqs="${line##*REQUIRES }"
         if [[ ! $hostonly ]] || grep -q "^$mod\$" "$DRACUT_KERNEL_MODALIASES"
         then
-            all_mods="$all_mods $reqs"
+            for module in $reqs;
+            do
+                is_built_in=$(awk 'BEGIN { FS = "/" } ; { print $NF }' $BUILT_IN_PATH | grep -wc $module.ko)
+                if [ $is_built_in -eq 0 ]
```

Comment:
How about:

`if grep -q "/$module.ko" $BUILT_IN_PATH; then ...`

which does not require awk (needs to be checked for and installed) nor use an (expensive) subshell? If you really want this to be thorough, you can use `grep -qe "/$module.ko$^"` to be extra-safe.

Reply:
Looks good to me, should be the following though:

`if ! grep -q "/$module.ko" $BUILT_IN_PATH`

as we are only interested in those that are not built-in.